### ISSUE TITLE
Error Importing tabbed TagSet.NullPointerException when importing tab separated files

### ISSUE THREAD

**Issue by GruffPrys** [webanno/webanno#281]
Error Importing tabbed TagSet.NullPointerException when importing tab separated files. These files imported successfully with Webanno 2 earlier this week. I'm using the standalone jars on Windows 10.

Simplest file is a utf-8 text file on Windows with the following content:

Noun
Verb
Adj
Adv
Pron
Det
Adp
Num
Conj
Prt
.
X

[universal_pos_tagset.txt](https://github.com/webanno/webanno/files/287986/universal_pos_tagset.txt)

**Comment by reckart** [comment]
Hard to say without a stack trace. 

Saying you tried WebAnno 2 before, I assume you are now trying WebAnno 3?

If you are trying standalone versions, WebAnno creates a temporary folder - you also need to delete that. It is a bit inconvenient but at least it is documented in the upgrade instructions:

> clean up the temporary installation before running the new version:
>
> On OS X: rm -R "$TMPDIR/winstoneEmbeddedWAR"
>
> On Linux: rm -R /tmp/winstoneEmbeddedWAR
>
> On Windows: remove the winstoneEmbeddedWAR that should be somewhere under C:\Users\<username>\AppData\Local\Temp

If that does not help, can you please provide the following:

* the exact version string from the bottom of the screen
* a stack trace

**Comment by reckart** [comment]
You are importing this as a tagset, right?

**Comment by GruffPrys** [comment]
Yes, I am trying Webanno 3. Sorry for not making that clear. Here's the full verson string:

Technische UniversitÃ¤t Darmstadt -- Computer Science Department -- WebAnno -- 3.0.0-beta-4 (2016-05-26 10:45:55, build 5c588cbbd8eb26e5b8e10ec8df88e90cadc85e1b)
US

I did follow the instructions and delete the winstoneEmbeddedWAR that was under C:\Users<username>\AppData\Local\Temp. However it's worth noting that I also had to delete the .webanno folder in my user folder before running the new for jar for it to work. Not doing so resulted in an "Error Code: 500" when accessing http://localhost:8080/.

Just to clarify, the Tagsets error occurs at http://localhost:8080/projectsetting.html on the Tagsets tab when using the Import Tagsets dialogue to import a text file of tags.

Is there a simple way fro me to generate a stack trace?

**Comment by reckart** [comment]
Looking at the code, the stack is actually currently swallowed - but I can reproduce the problem.

Please note that the *tab separated* format is not simply a list of tags:

```
<TAGSET-NAME>\t<TAGSET-DESCRIPTION>
<TAGSET-LANGUAGE>
<TAG1>\t<TAG1-DESCRIPTION>
<TAG2>\t<TAG2-DESCRIPTION>
...
```

E.g.

```
STTS	Stuttgart-Tübingen-Tag-Set \nGerman Part of Speech tagset STTS Tag Table
de	 
"	" '
$(	sonstige Satzzeichen; satzintern \nBsp: - [,]()
$,	Komma \nBsp: ,
$.	Satzbeendende Interpunktion \nBsp: . ? ! ; :   
...
```

We'll look into producing a better error message.

**Comment by reckart** [comment]
Please also note that we have change the default POS/DEP tagsets to the ones used by the Universal Dependencies project: https://github.com/webanno/webanno/issues/282

**Comment by GruffPrys** [comment]
You're absolutely right of course about the tab separated format. I'd forgotten that I'd corrected the tags using the GUI after importing them incorrectly on Webanno 2.

The POS and DEP tagsets were what I added, so I'm glad you're adding them as default tagsets. They're sure to be a good starting point for many.

I'm evaluating Webanno for use in a dependency annotation project I'm considering. By following the user guide I've managed to annotate a sentence's POSes, but I've not managed to work out how to annotate dependencies using the Universal DEP tagset. I'll ask for help on the Webanno Google Groups mailing list.
